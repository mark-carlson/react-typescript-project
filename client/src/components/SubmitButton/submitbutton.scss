.submit-btn {
    border: 1px solid black;
    padding: 20px 100px;
    background-color: white;
    font-size: 26px;
    &:focus {
        border: 1px solid red;
        outline: none;
    }
}