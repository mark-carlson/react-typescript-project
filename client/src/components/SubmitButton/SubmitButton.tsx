import React from 'react';
import './submitbutton.scss';

export type tSubmitButton = {
    className?: string;
    clickHandler: (val: string) => void;
    disabled?: boolean;
    text: string;
    value?: string;
};

export const SubmitButton = (props: tSubmitButton): JSX.Element => {
    const { className, clickHandler, disabled, text, value } = props;
    return (
        <>
            <button
                className={`submit-btn ${className ? className : ''}`}
                onClick={(): void => clickHandler(value)}
                disabled={disabled}
            >
                {text}
            </button>
        </>
    );
};
