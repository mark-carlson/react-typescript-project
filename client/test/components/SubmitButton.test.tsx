import React from 'react';
import { cleanup, render, getByText } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SubmitButton } from 'components/SubmitButton/SubmitButton';

afterAll(cleanup);

describe('<SubmitButton />', (): void => {
    const defaultProps = {
        clickHandler: jest.fn(),
        text: 'Click me',
    };
    it('should render correctly', (): void => {
        const { asFragment, container } = render(<SubmitButton {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
        const btn = getByText(container, defaultProps.text);
        btn.click();
        expect(defaultProps.clickHandler).toHaveBeenCalled();
    });
    it('should render correctly with a class', (): void => {
        const { asFragment } = render(<SubmitButton {...defaultProps} className="special-class" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
