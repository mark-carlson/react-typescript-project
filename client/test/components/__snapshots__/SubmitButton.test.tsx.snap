// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> should render correctly 1`] = `
<DocumentFragment>
  <button
    class="submit-btn "
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`<SubmitButton /> should render correctly with a class 1`] = `
<DocumentFragment>
  <button
    class="submit-btn special-class"
  >
    Click me
  </button>
</DocumentFragment>
`;
